arbiter/arbiter-core/src/main/java/org/deeplearning4j/arbiter/optimize/generator/genetic/crossover/ArithmeticCrossover.java

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************************
+ * Copyright (c) 2015-2018 Skymind, Inc.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.deeplearning4j.arbiter.optimize.generator.genetic.crossover;
+
+import org.apache.commons.math3.random.JDKRandomGenerator;
+import org.apache.commons.math3.random.RandomGenerator;
+import org.apache.commons.math3.random.SynchronizedRandomGenerator;
+import org.deeplearning4j.arbiter.optimize.generator.genetic.crossover.parentselection.RandomTwoParentSelection;
+import org.deeplearning4j.arbiter.optimize.generator.genetic.crossover.parentselection.TwoParentSelection;
+
+/**
+ * A crossover operator that linearly combines the genes of two parents. <br>
+ * When a crossover is generated (with a of probability <i>crossover rate</i>), each genes is a linear combination of the corresponding genes of the parents.
+ * <p>
+ * <i>t*parentA + (1-t)*parentB, where t is [0, 1] and different for each gene.</i>
+ * 
+ * @author Alexandre Boulanger
+ */
+public class ArithmeticCrossover extends TwoParentsCrossoverOperator {
+    private static final double DEFAULT_CROSSOVER_RATE = 0.85;
+
+    public static class Builder {
+        private double crossoverRate = DEFAULT_CROSSOVER_RATE;
+        private RandomGenerator rng;
+        private TwoParentSelection parentSelection;
+
+        /**
+         * The probability that the operator generates a crossover (default 0.85).
+         *
+         * @param rate A value between 0.0 and 1.0
+         */		
+        public Builder crossoverRate(double rate) {
+            if(rate < 0 || rate > 1.0) {
+                throw new IllegalArgumentException("Rate must be between 0.0 and 1.0");
+            }
```

Comment:
It's more a stylistic thing (I'll leave it up to you to change this sort of input validation if you want), but I tend to use the following (which is shorter and more informative for the user):
```
Preconditions.checkState(rate >= 0.0 && rate <= 1.0, "Rate must be between 0.0 and 1.0, got %s", rate);
```
That's the ND4J Preconditions class:
https://github.com/deeplearning4j/deeplearning4j/blob/master/nd4j/nd4j-common/src/test/java/org/nd4j/base/TestPreconditions.java
https://github.com/deeplearning4j/deeplearning4j/blob/master/nd4j/nd4j-common/src/main/java/org/nd4j/base/Preconditions.java

The ND4J preconditions class also formats arrays properly in the error string, and usually results in no object creation unless an exception is thrown.